backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/api/TriggerType.java

```diff
@@ -20,5 +20,5 @@
 package org.sonarsource.sonarlint.core.analysis.api;
 
 public enum TriggerType {
-  AUTO, FORCED
+  AUTO, FORCED, ANALYZE_FILES_AND_TRACK
```

Comment:
We can change the naming or document in code what is the rationale behind these 3 types.

Reply:
I change the names of the constants, and also tried to make things more explicit in the enum. Let me know what you think